### PR TITLE
Support probe feature in 0501-preview

### PR DESCRIPTION
**Pull request by Caoxuyang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring

File: src/spring/azext_spring/_params.py [hunk 1]
```diff
@@ -199,6 +199,21 @@ def load_arguments(self, _):
         c.argument('service', service_name_type)
         c.argument('name', name_type, help='Name of app.')
 
+    for scope in ['spring app create', 'spring app update', 'spring app deploy', 'spring app deployment create', 'spring app deployment update']:
+        with self.argument_context(scope) as c:
+            c.argument('enable_liveness_probe', arg_type=get_three_state_flag(),
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:212]:
```suggestion
            c.argument('enable_liveness_probe', arg_type=get_three_state_flag(), is_preview=True,
```

File: src/spring/azext_spring/_params.py [hunk 2]
```diff
@@ -199,6 +199,21 @@ def load_arguments(self, _):
...
+                       help='If false, will disable the liveness probe of the app instance', arg_group='App Customization')
+            c.argument('enable_readiness_probe', arg_type=get_three_state_flag(),
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:214]:
```suggestion
            c.argument('enable_readiness_probe', arg_type=get_three_state_flag(), is_preview=True,
```

File: src/spring/azext_spring/_params.py [hunk 3]
```diff
@@ -199,6 +199,21 @@ def load_arguments(self, _):
...
+                       help='If false, will disable the readiness probe of the app instance', arg_group='App Customization')
+            c.argument('enable_startup_probe', arg_type=get_three_state_flag(),
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:216]:
```suggestion
            c.argument('enable_startup_probe', arg_type=get_three_state_flag(), is_preview=True,
```

File: src/spring/azext_spring/_params.py [hunk 4]
```diff
@@ -199,6 +199,21 @@ def load_arguments(self, _):
...
+                       help='If false, will disable the startup probe of the app instance', arg_group='App Customization')
+            c.argument('liveness_probe_config', type=str,
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:218]:
```suggestion
            c.argument('liveness_probe_config', type=str, is_preview=True,
```

File: src/spring/azext_spring/_params.py [hunk 5]
```diff
@@ -199,6 +199,21 @@ def load_arguments(self, _):
...
+                       help='A json file path indicates the liveness probe config', arg_group='App Customization')
+            c.argument('readiness_probe_config', type=str,
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:220]:
```suggestion
            c.argument('readiness_probe_config', type=str, is_preview=True,
```

File: src/spring/azext_spring/_params.py [hunk 6]
```diff
@@ -199,6 +199,21 @@ def load_arguments(self, _):
...
+                       help='A json file path indicates the readiness probe config', arg_group='App Customization')
+            c.argument('startup_probe_config', type=str,
```
Comment (yuwzho) [inline src/spring/azext_spring/_params.py:222]:
```suggestion
            c.argument('startup_probe_config', type=str, is_preview=True,
```

File: src/spring/HISTORY.md [hunk 7]
```diff
@@ -6,6 +6,9 @@ Release History
 * Command `az spring update` has new argument "--ingress-read-timeout" to update ingress read timeout for Azure Spring Apps.
 * Command `az spring create` has new argument "--marketplace-plan-id" to purchase SaaS resource against given plan for Azure Spring Apps.
 * Command `az spring app deploy` and `az spring app deployment create` has new argument "--build-cpu" and "--build-memory" to set cpu and memory during build process.
+* Commands `az spring app create`, `az spring app update`, `az spring app deploy`, `spring app deployment create`
+and `spring app deployment update` have new arguments "enable_liveness_probe", "enable_readiness_probe", "enable_startup_probe", "liveness_probe_config", "readiness_probe_config", "startup_probe_config" to customize the probe settings of user applications
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:10]: > "enable_liveness_probe", "enable_readiness_probe", "enable_startup_probe", "liveness_probe_config", "readiness_probe_config", "startup_probe_config" 

Please use the correct parameter name

Comment (Caoxuyang) [reply src/spring/HISTORY.md]: Thx! Changed

**Comment by zhoxing-ms** [comment]
Do you need to release a new version for this PR? Or wait for other PRs to release the new version?

**Comment by Caoxuyang** [comment]
> Do you need to release a new version for this PR? Or wait for other PRs to release the new version?

Thank you for reminding!
We are still waiting for other PRs.

**Review by zhoxing-ms — APPROVED** [review]

**Comment by yuwzho** [comment]
#4844

**Pull request merged**